articles/tutorials/advanced/2d_shaders/07_sprite_vertex_effect/index.md

```diff
@@ -0,0 +1,863 @@
+		VertexShader = compile VS_SHADERMODEL MainVS();
+	}
+};
+```
+
+The shader will not compile yet, because the `VertexShaderOutput` has not been completely initialized. We need to replicate the `MatrixTransform` step to convert the vertices from world-space to clip-space. 
+
+Add the `MatrixTransform` shader parameter:
+
+```hlsl
+float4x4 MatrixTransform;
+```
+
+And then assign all of the output semantics in the vertex shader:
+
+```hlsl
+VertexShaderOutput MainVS(VertexShaderInput input) 
+{
+    VertexShaderOutput output;
+    output.Position = mul(input.Position, MatrixTransform);
+    output.Color = input.Color;
+    output.TextureCoordinates = input.TexCoord;
+    return output;
+}
+```
+
+To validate this is working, we should try to use the new effect. For now, we will experiment in the `TitleScene`. Create a class member for the new `Material`:
+
+```csharp
+// The 3d material  
+private Material _3dMaterial;
+```
+
+Load the shader using the hot reload system:
+
+```csharp
+// Load the 3d effect 
+_3dMaterial = Core.SharedContent.WatchMaterial("effects/3dEffect");
+```
+
+And then use the effect when drawing the title text:
+
+```csharp
+// Begin the sprite batch to prepare for rendering.  
+Core.SpriteBatch.Begin(  
+    samplerState: SamplerState.PointClamp,   
+    effect: _3dMaterial.Effect);
+```
+
+When the game runs, the text will be missing. This is because we never created a projection matrix to assign to the `MatrixTransform` shader parameter. Add this code when loading the material:
+
+```csharp
+// Load the 3d effect 
+_3dMaterial = Content.WatchMaterial("effects/3dEffect");
+_3dMaterial.IsDebugVisible = true;
+
+// this matrix code is taken from the default vertex shader code.
+Matrix.CreateOrthographicOffCenter(
+	left: 0, 
+	right: Core.GraphicsDevice.Viewport.Width, 
+	bottom: Core.GraphicsDevice.Viewport.Height, 
+	top: 0, 
+	zNearPlane: 0, 
+	zFarPlane: -1, 
+	out var projection);
+_3dMaterial.SetParameter("MatrixTransform", projection);
+```
+
+And now you should see the text normally again.
+
+![Figure 7.1: The main menu, but rendered with a custom vertex shader](./images/basic.png)
+
+### Making it Move
+
+As a quick experiment, we can show that the vertex shader can indeed modify the vertex positions further if we want to. For now, add a temporary shader parameter called `DebugOffset`:
+
+```hlsl
+float2 DebugOffset;
+```
+
+And change the vertex shader to add the `DebugOffset` to the `output.Position` after the clip-space conversion:
+
+```hlsl
+output.Position = mul(input.Position, MatrixTransform);  
+output.Position.xy += DebugOffset;
+```
+
+The sprites now move around as we adjust the shader parameter values. 
+![Figure 7.2: We can control the vertex positions](./gifs/basic.gif)
+
+It is important to build intuition for the different coordinate systems involved. Instead of adding the `DebugOffset` _after_ the clip-space conversion, if you try to add it _before_, like in the code below:
```

Comment:
Maybe also add an admonition to clear up the confusion, stating the differences in a single highlihgt